src/serde/tree_cache.rs

```diff
@@ -274,37 +276,24 @@ impl TreeCache {
                     let right = &self.node_entries[right_idx];
                     let serialized_length = 1 + left.serialized_length + right.serialized_length;
 
-                    let (hash, idx) = if let (Some(left_hash), Some(right_hash)) =
-                        (left.tree_hash, right.tree_hash)
```

Comment:
this check for whether the `tree_hash` is `None` needs a counterpart in the new way of doing this. As this is written now, the parent of the sentinel node will have a non-zero serialized length, but it's supposed to be "viral".